framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/KieServerDeploymentImpl.java

```diff
@@ -83,6 +92,41 @@ public String getSecureServiceName() {
         super.waitForScale();
         if (getInstances().size() > 0) {
             RouterUtil.waitForRouter(getUrl());
+            waitForSecureRouter();
+        }
+    }
+
+    private void waitForSecureRouter() {
+        LocalDateTime endTime = LocalDateTime.now().plus(Duration.of(5, ChronoUnit.SECONDS));
+        while (LocalDateTime.now().isBefore(endTime)) {
+            sendSecureRequiest();
+        }
+    }
+
+    private void sendSecureRequiest() {
+        final CredentialsProvider credentialsProvider = HttpsUtils.createCredentialsProvider(getUsername(),
+                getPassword());
+        try (CloseableHttpClient httpClient = HttpsUtils.createHttpClient(credentialsProvider)) {
```

Comment:
Just so you know: you can acquire multiple resources in try-with-resource statement like this:
```
try (Resource r1 = ...; Resource r2 = ...) {...}
```